### PR TITLE
Update COV code

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
Add options to select V4 and V5. Minor changes in headers to match spec. Remove support for V2. Set new default to V4 to match current default in assembler.

### CODE REVIEW

File: Tensile/KernelWriterAssembly.py [hunk 1]
```diff
@@ -225,16 +225,18 @@ def getCompileArgs(self, sourceFileName, objectFileName, *moreArgs, isa=None, wa
...
-    archHasV3 = globalParameters["AsmCaps"][isa]["HasCodeObjectV3"]
-
     launcher = shlex.split(os.environ.get('Tensile_ASM_COMPILER_LAUNCHER', ''))
 
     rv = launcher + [globalParameters['AssemblerPath'],
           '-x', 'assembler',
           '-target', 'amdgcn-amd-amdhsa']
 
-    if archHasV3:
-      rv += ['-mcode-object-version=2' if globalParameters["CodeObjectVersion"] == "V2" else '-mcode-object-version=4']
```
Comment (nakajee) [inline Tensile/KernelWriterAssembly.py:237]: Why -mcode-object-version was 4 for V3 before change?
Is 3 the correct value?

Comment (AlexBrownAMD) [reply Tensile/KernelWriterAssembly.py:237]: No, 3 means COV3, we just had V4 partially hacked in before

File: Tensile/Components/Signature.py [hunk 2]
```diff
@@ -380,7 +146,15 @@ def __call__(self, writer):
         kStr += "---\n"
         kStr += "amdhsa.version:\n"
         kStr += "  - 1\n"
-        kStr += "  - 0\n"
+        if kernel["CodeObjectVersion"] == "V3":
```
[the remark this answers is not in the document]
Comment (AlexBrownAMD) [reply Tensile/Components/Signature.py:149]: I think it happens in SolutionStructs init. COV gets added to state there, and solution is used to create kernel objects that get passed to code gen

**Comment by nakajee** [comment]
Thanks for your comments.
I do not have further comments.
I would like to wait for comments from other reviewers.

**Review by benjaminulmer — APPROVED** [review]
Code looks good to me.

Conflicts are from merging in #1605. Take care not to squash those changes when fixing the conflics.

File: Tensile/Components/Signature.py [hunk 3]
```diff
@@ -23,248 +23,14 @@
...
-        kStr += "      NumVGPRs:             %u%s" % ( totalVgprs, writer.endLine )
-        kStr += "      MaxFlatWorkGroupSize: %u%s" % ( kernel["SubGroup0"] * kernel["SubGroup1"] * kernel["LocalSplitU"], writer.endLine )
-        kStr += ".end_amd_amdgpu_hsa_metadata\n"
-
-        return kStr
-
-
-class SignatureCOV3(Signature):
-    kernel = {"CodeObjectVersion": "V3"}
+# Creates kernel header, compatible with code object version 3 and up. COV2 no longer supported.
```
[the remark this answers is not in the document]
Comment (AlexBrownAMD) [reply Tensile/Components/Signature.py]: Updated

**Comment by AlexBrownAMD** [comment]
> Conflicts are from merging in #1605. Take care not to squash those changes when fixing the conflics.

When merging I just used the versions from your commit, since it looks like they removed any specific CodeObjectVersion form the yaml and default is the desired behavior.

**Comment by nakajee** [comment]
It looks like precheckin build failed with the following error message.
  Tensile::FATAL: Unknown CodeObjectVersion V3

**Comment by AlexBrownAMD** [comment]
> It looks like precheckin build failed with the following error message. Tensile::FATAL: Unknown CodeObjectVersion V3

Caused by new test files that were added in latest merge. They've been updated now.

**Pull request merged**